float64[] x_start
float64[] x_goal
float64 c_max
float64 c_min